FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,259 @@
+﻿//
+// This file is part of the Frenetic Game Engine, created by Frenetic LLC.
+// This code is Copyright (C) Frenetic LLC under the terms of a strict license.
+// See README.md or LICENSE.txt in the FreneticGameEngine source root for the contents of the license.
+// If neither of these are available, assume that neither you nor anyone other than the copyright holder
+// hold any right or permission to use this software until such time as the official license is identified.
+//
+
+using FGECore.ModelSystems;
+using System;
+using System.Collections.Generic;
+using OpenTK.Mathematics;
+using FGECore.CoreSystems;
+
+namespace FGEGraphics.GraphicsHelpers.Models;
+
+/// <summary>Generates shapes and returns them as a model.</summary>
+public static class ShapeGenerators
+{
+    /// <summary>
+    /// Generates a 3D sphere model.
+    /// Note that the stacks should be lower than the slices to avoid incorrect normals a good example is 10 stacks and 60 slices.
+    /// </summary>
+    public static Model GenerateSphere(float radius, uint stacks, uint slices, ModelEngine modelEngine)
+    {
+        if (stacks > slices)
+        {
+            Logs.Warning("Sphere has more stacks than slices, this may result in incorrect normals.");
+        }
+        uint vertexCount = (stacks + 1) * (slices + 1);
```

Comment:
Re-noting as prior comment was lifted that the `+ 1`s still don't make sense here.

Reply:
Previously it was to encompass the sphere fully but that is no longer the case. I also removed it for the donut but the cylinder still needs it because it is in use by the indices and if you don't do that you get this:

![bandicam 2024-03-19 20-28-05-515](https://github.com/FreneticLLC/FreneticGameEngine/assets/97306922/bef9a679-41e8-4f7a-b877-e1f1d77161cd)

Reply:
Again, you can just... `- 1` on specific parts that need to be 1 smaller